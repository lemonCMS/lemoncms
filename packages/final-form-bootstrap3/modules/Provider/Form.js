import React from "react";
import PropTypes from "prop-types";
import { Form as FinalForm } from "react-final-form";
import arrayMutators from "final-form-arrays";
import ContextWrapper from "../Context/ContextWrapper";

const Form = props => {
  const { children, className, debug, readOnly, listen, ...formProps } = props;

  return (
    <FinalForm
      {...formProps}
      mutators={{
        ...arrayMutators
      }}
      render={({ handleSubmit, ...rest }) => {
        return (
          <ContextWrapper
            {...rest}
            debug={debug}
            listen={listen}
            readOnly={readOnly}
          >
            <form onSubmit={handleSubmit} className={className}>
              {children}
            </form>
          </ContextWrapper>
        );
      }}
    />
  );
};

Form.propTypes = {
  className: PropTypes.string,
  validate: PropTypes.func,
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
  debug: PropTypes.bool,
  readOnly: PropTypes.bool,
  listen: PropTypes.func,
  initialValues: PropTypes.objectOf(
    PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
      PropTypes.array,
      PropTypes.object,
      PropTypes.bool
    ])
  )
};

Form.defaultProps = {
  className: "",
  validate: () => {},
  listen: () => {},
  debug: false,
  readOnly: false,
  initialValues: {}
};

export default Form;
