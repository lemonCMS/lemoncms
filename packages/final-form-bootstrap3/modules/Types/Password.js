import React from "react";
import PropTypes from "prop-types";
import { FormControl } from "react-bootstrap";
import context from "../decorators/context";
import fieldGroup from "./fieldGroup";

const Password = props => {
  const { input, isDisabled } = props;
  return <FormControl {...input} type={"password"} disabled={isDisabled} />;
};

Password.propTypes = {
  input: PropTypes.shape({
    name: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    onBlur: PropTypes.func.isRequired,
    onFocus: PropTypes.func.isRequired,
    value: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.bool,
      PropTypes.number
    ])
  }),
  placeholder: PropTypes.string,
  label: PropTypes.string,
  help: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired
};

Password.defaultProps = {
  input: {},
  label: null,
  help: null,
  placeholder: null
};

export default context()(fieldGroup(Password));
