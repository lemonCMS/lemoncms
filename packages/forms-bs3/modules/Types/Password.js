import React from "react";
import PropTypes from "prop-types";
import { FormControl } from "react-bootstrap-v3";
import context from "../decorators/context";
import fieldGroup from "./fieldGroup";

const Password = props => {
  const { input } = props;
  return <FormControl {...input} type={"password"} />;
};

Password.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]),
  placeholder: PropTypes.string,
  label: PropTypes.string,
  help: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired
};

Password.defaultProps = {
  input: {},
  label: null,
  help: null,
  placeholder: null
};

export default context(fieldGroup(Password));
