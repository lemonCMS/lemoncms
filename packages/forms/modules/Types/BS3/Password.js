import React from "react";
import PropTypes from "prop-types";
import {
  FormGroup,
  ControlLabel,
  FormControl,
  HelpBlock
} from "react-bootstrap-v3";
import context from "../../decorators/context";
import field from "../../decorators/field";

const Password = props => {
  const { input, placeholder, computedInvalid, label, help } = props;

  return (
    <FormGroup validationState={computedInvalid ? "error" : null}>
      {label && <ControlLabel>{label}</ControlLabel>}
      <FormControl type="password" placeholder={placeholder} {...input} />
      <FormControl.Feedback />
      {help && <HelpBlock>{help}</HelpBlock>}
    </FormGroup>
  );
};

Password.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]),
  placeholder: PropTypes.string,
  label: PropTypes.string,
  help: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired
};

Password.defaultProps = {
  input: {},
  label: null,
  help: null,
  placeholder: null
};

export default context(field(Password));
