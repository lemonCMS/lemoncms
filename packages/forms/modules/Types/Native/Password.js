import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import context from '../../decorators/context';
import field from '../../decorators/field';

const Password = props => {
  const {
    input, placeholder, formControl, computedInvalid
  } = props;
  return (
    <input
      type={'password'}
      placeholder={placeholder}
      className={classNames('ffc-input', { 'is-invalid': computedInvalid }, formControl)}
      {...input}
    />
  );
};

Password.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]),
  placeholder: PropTypes.string,
  formControl: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired
};

Password.defaultProps = {
  input: {},
  placeholder: null,
  formControl: null
};

export default context(field(Password));
