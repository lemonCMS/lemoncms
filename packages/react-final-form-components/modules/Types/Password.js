import React from "react";
import PropTypes from "prop-types";
import { FormControl } from "react-bootstrap";
import context from "../decorators/context";
import fieldGroup from "./fieldGroup";

const Password = props => {
  const { input, isDisabled, placeholder, readOnly } = props;

  return (
    <FormControl
      {...input}
      type={"password"}
      disabled={isDisabled}
      placeholder={placeholder}
      readOnly={readOnly}
    />
  );
};

Password.propTypes = {
  input: PropTypes.shape({
    name: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    onBlur: PropTypes.func.isRequired,
    onFocus: PropTypes.func.isRequired,
    value: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.bool,
      PropTypes.number
    ])
  }),
  disabled: PropTypes.func,
  readOnly: PropTypes.bool,
  isDisabled: PropTypes.bool,
  placeholder: PropTypes.string,
  label: PropTypes.string,
  help: PropTypes.string,
  addOn: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  addOnBefore: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  mandatory: PropTypes.bool,
  computedInvalid: PropTypes.bool.isRequired,
  layout: PropTypes.shape({
    label: PropTypes.object,
    field: PropTypes.object
  })
};

Password.defaultProps = {
  input: {},
  label: null,
  help: null,
  addOn: null,
  addOnBefore: null,
  placeholder: null,
  disabled: null,
  isDisabled: false,
  mandatory: false,
  layout: null
};

export default context()(fieldGroup(Password));
