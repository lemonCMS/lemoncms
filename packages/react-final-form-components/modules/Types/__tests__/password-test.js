import renderer from "react-test-renderer";
import { mount } from "enzyme";
import Form from "../../Provider/Form";
import Password from "../Password";
import React from "react";

describe("BS3 Password type", () => {
  it("should render input password", () => {
    const tree = renderer.create(
      <Form
        onSubmit={() => {}}
        validate={() => ({ password: "Oepsie daisy" })}
        initialValues={{ password: "password 0091" }}
      >
        <Password
          name={"password"}
          placeholder={"a nice placeholder"}
          help={"What was your mothers maiden name?"}
        />
      </Form>
    );
    const output = tree.toJSON();
    expect(output).toMatchSnapshot();
  });

  it("should render input password enzyme", () => {
    const callback = jest.fn();
    callback.mockReturnValueOnce({ password: "password 0091" });
    const component = mount(
      <Form
        onSubmit={() => {}}
        validate={callback}
        initialValues={{ password: "password 0091" }}
      >
        <Password
          name={"password"}
          placeholder={"a nice placeholder"}
          help={"What was your mothers maiden name?"}
        />
        <button type={"submit"}>submit</button>
      </Form>
    );

    const input = component.find("input");
    expect(input).toBeDefined();

    const { value, name } = input.props();
    expect(value).toEqual("password 0091");
    expect(name).toEqual("password");

    /**
     * Help block is set
     */
    const helpBlock = component.find(".help-block");
    expect(helpBlock.text()).toEqual("What was your mothers maiden name?");

    /**
     * Test that error class will be set
     */
    expect(component.find(".has-error").length).toEqual(0);
    component.find("button").simulate("submit");
    expect(callback).toHaveBeenCalled();
    expect(component.find(".has-error").length).toEqual(1);
  });
});
