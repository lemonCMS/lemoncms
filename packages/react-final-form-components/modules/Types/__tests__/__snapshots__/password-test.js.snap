// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BS3 Password type should render input password 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="form-group has-feedback"
  >
    <div
      className="col-sm-8"
    >
      <input
        className="form-control"
        disabled={null}
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="password"
        value="password 0091"
      />
    </div>
    <span
      className="help-block"
    >
      What was your mothers maiden name?
    </span>
    <div
      className="clearfix"
    />
  </div>
</form>
`;
